chapters/1550 Protocols.md

```diff
@@ -0,0 +1,1111 @@
+      <span class="arrow"> → </span> 
+      <a href="#identifier">identifier</a>
+    </p>
+  </div>
+  <div class="syntax-group">
+    <p class="syntax-def">
+      <span class="name" id="type-annotation">type-annotation</span>
+      <span class="arrow"> → </span> 
+      <code>:</code>
+      <a href="#attributes">attributes</a><sub class="loosen">opt</sub>
+      <code>inout</code><sub>opt</sub>
+      <a href="#type">type</a>
+    </p>
+  </div>
+  <div class="syntax-group">
+    <p class="syntax-def">
+      <span class="name" id="getter-setter-keyword-block">
+      getter-setter-keyword-block</span>
+      <span class="arrow"> → </span> 
+      <code>{</code>
+      <a href="#getter-keyword-clause">getter-keyword-clause</a>
+      <a href="#setter-keyword-clause">
+      setter-keyword-clause</a><sub class="loosen">opt</sub>
+      <code>}</code>
+    </p>
+    <p class="syntax-def">
+      <span class="name" id="getter-setter-keyword-block">
+      getter-setter-keyword-block</span>
+      <span class="arrow"> → </span> 
+      <code>{</code>
+      <a href="#setter-keyword-clause">setter-keyword-clause</a>
+      <a href="#getter-keyword-clause">getter-keyword-clause</a>
+      <code>}</code>
+    </p>
+    <p class="syntax-def">
+      <span class="name" id="getter-keyword-clause">getter-keyword-clause</span>
+      <span class="arrow"> → </span> 
+      <a href="#attributes">attributes</a><sub class="loosen">opt</sub>
+      <a href="#mutation-modifier">
+      mutation-modifier</a><sub class="loosen">opt</sub>
+      <code>get</code>
+    </p>
+    <p class="syntax-def">
+      <span class="name" id="setter-keyword-clause">setter-keyword-clause</span>
+      <span class="arrow"> → </span> 
+      <a href="#attributes">attributes</a><sub class="loosen">opt</sub>
+      <a href="#mutation-modifier">
+      mutation-modifier</a><sub class="loosen">opt</sub>
+      <code>set</code>
+    </p>
+  </div>
+  <div class="syntax-group">
+    <p class="syntax-def">
+      <span class="name" id="mutation-modifier">mutation-modifier</span>
+      <span class="arrow"> → </span> 
+      <code>mutating</code>
+      |
+      <code>nonmutating</code>
+    </p>
+  </div>
+</div>
+
+<p class="draft item"> </p>
+[what it is]
+
+<p class="draft item"> </p>
+[how is it declared]
+
+
+#### Method or Operator Function
+
+<div class="admonition grammar">
+  <p class="admonition-title">Grammar of a protocol method declaration</p>
+  <div class="syntax-group">
+    <p class="syntax-def">
+      <span class="name" id="protocol-method-declaration">
+      protocol-method-declaration</span>
+      <span class="arrow"> → </span> 
+      <a href="#function-head">function-head</a>
+      <a href="#function-name">function-name</a>
+      <a href="#generic-parameter-clause">
+      generic-parameter-clause</a><sub class="loosen">opt</sub>
+      <a href="#function-signature">function-signature</a>
+    </p>
+  </div>
+</div>
```

Comment:
@dabrahams: Regarding usage of html rather than markup language for grammar. 

Reasons html tags are necessary:
1. To name syntax definitions for hyperlinking.
2. To get subscripting for "opt".

Reasons html tags are desirable:
1. To achieve pleasant formatting that makes for easier reading of a definition.

Reasons html tags are undesirable:
1. Visual clutter when editing.
1. Difficult to formulate from scratch.

Reasons why the visual clutter and difficult formulation are ok:
1. The html block is standardized, and easily cut, pasted and adapted.
1. The clutter isn't so bad when compared to the markup language version, which would look like:

```
GRAMMAR OF A PROTOCOL METHOD DECLARATION

<span class="name" id="protocol-method-declaration">
       protocol-method-declaration</span>
-> [function-head](#function-head) [function-name](#function-name) 
[generic-parameter-clause](#generic-parameter-clause)<sub>opt</sub>
[function-signature](#function-signature)
```

To my eye, the html tag version is easier to read and easier to write.

Thoughts?